usr.bin/paste/paste.c

```diff
@@ -98,41 +114,22 @@ main(int argc, char *argv[])
 		delim = tab;
 	}
 
-	if (seq)
-		rval = sequential(argv);
-	else
-		rval = parallel(argv);
-	exit(rval);
-}
-
-typedef struct _list {
-	struct _list *next;
-	FILE *fp;
-	int cnt;
-	char *name;
-} LIST;
-
-static int
-parallel(char **argv)
-{
-	LIST *lp;
-	int cnt;
-	wint_t ich;
-	wchar_t ch;
-	char *p;
-	LIST *head, *tmp;
-	int opencnt, output;
-
-	for (cnt = 0, head = tmp = NULL; (p = *argv); ++argv, ++cnt) {
+	for (filecnt = 0, head = tmp = NULL; *argv; ++argv, ++filecnt) {
+		failed = 0;
 		if ((lp = malloc(sizeof(LIST))) == NULL)
 			err(1, NULL);
-		if (p[0] == '-' && !p[1])
+		if ((*argv)[0] == '-' && !(*argv)[1])
```

Comment:
Maybe just do argv[0][0] == '-' && argv[0][1] == '\0' instead of using a pointer (or !argv[0][1] is okay too).

Reply:
While both approaches likely generate the very same code, I opted for the pointer dereference since per C23 6.3.2.1p3, to my understanding, the expression is automatically converted from an array to a pointer anyway.

Reply:
I referenced C23 as it's the latest approved draft, but section 6.3.2.1p3 is the same in C99 as well. Regarding readability, I have a different opinion, but I’m completely fine if you (or anyone else committing the PR later) decide to change it.